pkg/apis/metrics/prometheus_writer.go

```diff
@@ -238,5 +238,8 @@ func convertMetricName(in string) string {
 		out = append(out, unicode.ToLower(runes[i]))
 	}*/
 
-	return strings.Replace(in, ".", "_", -1)
+//	return strings.Replace(in, ".", "_", -1)
+	tmp := strings.Replace(in, ".", "_", -1)
```

Comment:
```golang
return strings.NewReplacer("-","_",".","_").Replace()
```
will better

Reply:
Very good. I have made changes and submitted them again. I hope I can make changes in the next version. Thank you.